runtime/jcl/common/reflecthelp.c

```diff
@@ -453,7 +460,11 @@ classForSignature(struct J9VMThread *vmThread, U_8 **sigDataPtr, struct J9ClassL
 
 	/* Non-array case */
 	switch (c) {
-	case 'L': {
+	case 'L':
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
```

Comment:
at this point we are not sure if Q and L types will share the same ramClass or not. So for now leave this out but add a comment saying this needs to be revisited